Patch-Source: https://github.com/rsyslog/rsyslog/pull/6410
--
From a2d815178cc7238c932d6b82e40815e4a22ad77e Mon Sep 17 00:00:00 2001
From: Rainer Gerhards <rgerhards@adiscon.com>
Date: Fri, 2 Jan 2026 11:46:45 +0100
Subject: [PATCH] tcpsrv: restore default stack size assignment

Worker threads need to be assigned the "rsyslog default stack size".
This is especially an issue with musl lib, which has a very small stack
size. As such, rsyslog can segfault without specifying our expected
stack size.

closes https://github.com/rsyslog/rsyslog/issues/6357
---
 runtime/tcpsrv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/runtime/tcpsrv.c b/runtime/tcpsrv.c
index d9592a74e9..ee3d2d34a9 100644
--- a/runtime/tcpsrv.c
+++ b/runtime/tcpsrv.c
@@ -1202,7 +1202,7 @@ static rsRetVal ATTR_NONNULL() startWrkrPool(tcpsrv_t *const pThis) {
     /* Spawn workers. */
     pThis->currWrkrs = 0;
     for (unsigned i = 0; i < queue->numWrkr; ++i) {
-        if (pthread_create(&queue->wrkr_tids[i], NULL, wrkr, pThis) != 0) {
+        if (pthread_create(&queue->wrkr_tids[i], &default_thread_attr, wrkr, pThis) != 0) {
             iRet = RS_RET_ERR;
             break;
         }
